### PR TITLE
tsuba: RDGPartHeader fix partition file regex

### PR DESCRIPTION
**Pull request by pyrito**
This PR intends to fix a bug found with the deployment. There was an issue where every partition file was being copied over with the same node id. This was because there was an issue with the regex parsing which caused every partition file to be copied over with the same node id of 0, hence the copies were getting butchered. I also wrote a test to catch this edge case.

Linked PR: https://github.com/KatanaGraph/katana-enterprise/pull/1980

### CODE REVIEW

File: libtsuba/src/RDGPartHeader.cpp [hunk 1]
```diff
@@ -74,7 +74,7 @@ namespace tsuba {
 
 // Regex for partition files
 const std::regex kPartitionFile(
-    "part_(?:(vers[0-9A-Za-z_]+))_(?:(rdg[0-9A-Za-z-]*))_(?:(node[0-9]*))$");
+    "part_vers(?:([0-9A-Za-z_]+))_(?:(rdg[0-9A-Za-z-]*))_node(?:([0-9]*))$");
```
Comment (tylershunt) [inline libtsuba/src/RDGPartHeader.cpp]: this regex is still weird to me; why do we:
 * Allow arbitrary characters in the version but not in the node number?
 * why are all these non match groups used?
 * Why is the node id allowed to be empty?
 I think a clearer regex would be:
```
"part_vers([0-9]+)_(rdg[0-9A-Za-z-]*)_node([0-9]+)$"
```

**Review by tylershunt — APPROVED** [review]

File: libtsuba/src/RDGPartHeader.cpp (hunk 1 above)
Comment (pyrito) [reply libtsuba/src/RDGPartHeader.cpp]: I'll be honest, I'm pretty bad with regex. I kind of went with something that got the job done to some extent. I think your restrictions on the regex make sense. I'll implement this.

File: libtsuba/src/tsuba.cpp [hunk 2]
```diff
@@ -220,6 +220,7 @@ tsuba::CreateSrcDestFromViewsForCopy(
     auto rdg_manifest_res =
         tsuba::RDGManifest::Make(uri, rdg_view.view_type, version);
     if (!rdg_manifest_res) {
+      KATANA_LOG_WARN("not a valid manifest file, uri: {}", uri.string());
```
Comment (witchel) [inline libtsuba/src/tsuba.cpp:223]: Would we be grabbing a Tracer object here instead of WARN?  I'm really not sure.

**Review by witchel — APPROVED** [review]
Trust in regex.

File: libtsuba/src/tsuba.cpp (hunk 2 above)
Comment (pyrito) [reply libtsuba/src/tsuba.cpp:223]: I didn't think this was appropriate for a Tracer print so I stuck with a warning log.
